src/lib/u2.js

```diff
@@ -0,0 +1,1419 @@
+/**
+ * @license
+ * Copyright 2016 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/*
+TODO:
+ - Fix handling of Slots that return arrays.
+ - Add support for FOAM1 style dynamic functions?
+ - Properly handle insertBefore_ of an element that's already been inserted?
+ - Fix ExpressionSlot firing too many property change events
+*/
+
+foam.CLASS({
+  package: 'foam.u2',
+  name: 'EID',
+  extends: 'Property',
+  documentation: 'Describes a property used to store a DOM element id.',
+
+  constants: {
+    __ID__: [ 0 ],
+    NEXT_ID: function() {
+      return 'u2v' + this.__ID__[ 0 ]++;
+    }
+  },
+
+  properties: [
+    {
+      name: 'factory',
+      value: function() { return foam.u2.EID.NEXT_ID(); }
```

Comment:
This could probably be changed to just use 'u2v' + foam.next$UID();

No need to have multiple unique id generators.